src/Element/MultiCheckbox.php

```diff
@@ -99,17 +99,16 @@ public function setOptions(iterable $options)
         return $this;
     }
 
-    /**
-     * Set a single element attribute
-     *
-     * @param  mixed  $value
-     * @return $this
-     */
+    /** @inheritDoc */
     public function setAttribute(string $key, $value)
     {
-        // Do not include the options in the list of attributes
-        // TODO: Deprecate this
-        if ($key === 'options') {
+        /** @psalm-suppress DocblockTypeContradiction */
+        if ($key === 'options' && is_iterable($value)) {
+            trigger_error(
+                'Providing multi-checkbox value options via attributes is deprecated and will be removed in '
+                . 'version 4.0 of this library',
```

Comment:
Does building the element from an array spec still work, after removing this?

Reply:
@gsteel seems a prudent guy:

https://github.com/laminas/laminas-form/pull/245/files#diff-9ed265bc15aed45c500eadeea5fec4faa6af32a6800c80a282eac53c51b604d4R188-R216

Reply:
Was more thinking about the really wonky early-laminas "let's make an entire form from a single array" kinda thing :D

Reply:
Yes, it does still work - The factory tests _were_ showing this, but I did adjust this test in https://github.com/laminas/laminas-form/pull/245/commits/da4a9f0f8e17adaf20bfbd14d53c590f54768ab8 so that the deprecated version is no longer used.